mt-kahypar/partition/coarsening/multilevel_coarsener.h

```diff
@@ -462,8 +464,6 @@ class MultilevelCoarsener : public ICoarsener,
       ++contracted_nodes;
       success = true;
     }
-    _matching_partner[u] = u;
-    _matching_state[u] = STATE(MatchingState::MATCHED);
```

Comment:
Is there no need to set the matching partner?

Reply:
These writes are fully redundant. They already happen in `matchVertices`, and doing it here again (or rather: already) also results in the wrong ordering compared to what is expected by the assertions.